src/Symfony/Component/Uid/ArgumentResolver/UidValueResolver.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Uid\AbstractUid;
+use Symfony\Component\Uid\Attribute\Uid;
+
+final class UidValueResolver implements ArgumentValueResolverInterface
+{
+    private string $defaultFormat;
+
+    public function __construct(string $defaultFormat = Uid::FORMAT_CANONICAL)
+    {
+        $this->defaultFormat = $defaultFormat;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        if ($argument->isVariadic() || !\is_string($request->attributes->get($argument->getName())) || null === $argument->getType()) {
+            return false;
+        }
+
+        try {
+            $refl = new \ReflectionClass($argument->getType());
```

Comment:
is_subclass_of() instead of using reflection?

Reply:
I used reflection to check that the class is not abstract. Do you know a better way?

Reply:
what if we don't care?

Reply:
Returning true here means that it will fail 100% in `resolve()`. I would have to catch the `\Error` like I did in https://github.com/symfony/symfony/pull/44600/files#diff-080936a60ff65508dfa5c42d744d610331a8682724a3d10dbb43bb6817e50b41R76. Is that better than doing one reflection?

Reply:
Let's just use `is_subclass_of` and let it fail in resolve if the user makes the mistake of using an abstract class.